wafv2/rate-based-acl/main.tf

```diff
@@ -0,0 +1,107 @@
+resource "aws_wafv2_web_acl" "rate_based" {
+  name        = "rate-based"
+  description = "Rate based ACL"
+  scope       = "REGIONAL"
+
+  dynamic "default_action" {
+    for_each = var.action_default == "allow" ? ["allow"] : []
+    content {
+      allow {}
+    }
+  }
+
+  dynamic "default_action" {
+    for_each = var.action_default == "block" ? ["block"] : []
+    content {
+      allow {}
+    }
+  }
+
+
+  rule {
+    name     = "x-forwarded-for"
+    priority = 1
+
+    dynamic "action" {
+      for_each = var.action_x-forwarded-for == "allow" ? ["allow"] : []
+      content {
+        allow {}
+      }
+    }
+
+    dynamic "action" {
+      for_each = var.action_x-forwarded-for == "block" ? ["block"] : []
+      content {
+        block {}
+      }
+    }
+
+    dynamic "action" {
+      for_each = var.action_x-forwarded-for == "count" ? ["count"] : []
+      content {
+        count {}
+      }
+    }
+
+    dynamic "action" {
+      for_each = var.action_x-forwarded-for == "captcha" ? ["captcha"] : []
+      content {
+        captcha {}
+      }
+    }
+
+    statement {
+      rate_based_statement {
+
+        limit              = var.rate_based_limit
+        aggregate_key_type = "FORWARDED_IP"
+
+        forwarded_ip_config {
+          fallback_behavior = "NO_MATCH"
+          header_name       = "X-Forwarded-For"
+        }
+
+        dynamic "scope_down_statement" {
+          for_each = length(var.paths) > 0 ? ["scope_down_statement"] : []
+          content {
+            or_statement {
+              dynamic "statement" {
+                for_each = { for idx, path in var.paths: (idx) => path }
+                content {
+                  byte_match_statement {
+                    positional_constraint = "CONTAINS"
+                    search_string         = statement.value
+                    text_transformation {
+                      priority = tonumber(scope_down_statement.key)
```

Comment:
isn't that always 0 in our case ? if so, it would be more readable IMO to put explicitly 0 here

Reply:
My bad, this is wrong, mean to be different, will update